chitchat/src/state.rs

```diff
@@ -24,14 +24,29 @@ pub struct NodeState {
     chitchat_id: ChitchatId,
     heartbeat: Heartbeat,
     key_values: BTreeMap<String, VersionedValue>,
-    max_version: Version,
     #[serde(skip)]
     listeners: Listeners,
+    max_version: Version,
     // This is the maximum version of the last tombstone GC.
     //
-    // After we GC the tombstones, we cannot safely do replication with
-    // nodes that are asking for a diff from a version lower than this.
+    // Due to the garbage collection of tombstones, we cannot
+    // safely do replication with nodes that are asking for a
+    // diff from a version lower than this.
+    //
+    // `last_gc_version` expresses the idea: what is the oldest version from which I can
+    // confidently emit delta from. The reason why we update it here, is
+    // because a node that was just reset or just joined the cluster will get updates
+    // from another node that are actually only make sense in the context of the
+    // emission of delta from a `last_gc_version`.
     last_gc_version: Version,
+    // A proper interpretation of `max_version` and `last_gc_version` is the following:
+    // The state contains exactly:
+    // - all of the (non-deleted) key values present at snapshot `max_version`.
+    // - all of the tombstones of the entry that were deleted between (`last_gc_version`,
+    //   `max_version]`.
+    //
```

Comment:
Can we add that we always have last_gc_version <= max_version?

Reply:
No... But I can add the opposite because this is not true :)